### PR TITLE
refactor: Eliminate synchronization conflicts between tsm and tsm2, and between iterator v1 and iterator v2.

### PR DESCRIPTION
**Pull request by liikeuforever**
# Required checklist
- [ ] Sample config files updated (`config`,`meta/config` and `default config`) 
- [ ] If there are user-facing changes, the documentation needs to be updated prior to approving the PR( [Link]() )
- [ ] If there are any breaking changes to public APIs, please add the `api change` label.
- [ ] Signed [CLA](https://cla-assistant.io/cnosdb/cnosdb) (if not already signed)

# Which issue does this PR close?

[//]: # (We generally require a GitHub issue to be filed for all bug fixes and enhancements and this helps us generate change logs for our releases. You can link an issue to this PR using the GitHub syntax. For -- example `Closes #123` indicates that this PR will close issue #123.)

Related #.

# Rationale for this change

refactor: Eliminate synchronization conflicts between tsm and tsm2, and between iterator v1 and iterator v2.

[//]: # (Why are you proposing this change? If this is already explained clearly in the issue then this section is not needed.
 Explaining clearly why changes are proposed helps reviewers understand your changes and offer better suggestions for fixes.)

# Are there any user-facing changes?

[//]: # (There is no need to duplicate the description in the issue here, but it is sometimes worth providing a summary of the individual changes in this PR.)

### CODE REVIEW

File: tskv/src/tseries_family.rs [hunk 1]
```diff
@@ -506,17 +505,15 @@ impl Version {
         vec![]
     }
 
-    pub async fn get_tsm_reader2(&self, path: impl AsRef<Path>) -> Result<Arc<TSM2Reader>> {
```
Comment (Subsegment) [inline tskv/src/tseries_family.rs:509]: here

File: tskv/src/compaction/flush.rs [hunk 2]
```diff
@@ -50,7 +50,7 @@ impl FlushTask {
         }
     }
 
-    pub async fn tsm2_run(
```
Comment (Subsegment) [inline tskv/src/compaction/flush.rs:53]: just run

File: tskv/src/tsm/mod.rs [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
 const INDEX_META_SIZE: usize = 11;
 const BLOCK_META_SIZE: usize = 44;
 const BLOOM_FILTER_SIZE: usize = 64;
-const BLOOM_FILTER_BITS: u64 = 512; // 64 * 8
-const FOOTER_SIZE: usize = BLOOM_FILTER_SIZE + 8; // 72
+const BLOOM_FILTER_BITS: u64 = 512;
+// 64 * 8
+const FOOTER_SIZE: usize = 129;
+
+pub type TsmWriteData = BTreeMap<String, BTreeMap<SeriesId, (SeriesKey, DataBlock2)>>; // (table, (series_id, pages))
```
Comment (Subsegment) [inline tskv/src/tsm/mod.rs:28]: DataBlock2

Comment (liikeuforever) [reply tskv/src/tsm/mod.rs]: Done.

**Comment by Subsegment** [comment]
why format generated file

**Comment by liikeuforever** [comment]
why format generated file

Done

File: tskv/src/reader/column_group/mod.rs [hunk 4]
```diff
@@ -19,8 +19,8 @@ use super::{
     BatchReader, BatchReaderRef, SchemableTskvRecordBatchStream,
     SendableSchemableTskvRecordBatchStream,
 };
-use crate::tsm2::page::{ColumnGroup, PageWriteSpec};
-use crate::tsm2::reader::TSM2Reader;
+use crate::tsm::page::{ColumnGroup, PageWriteSpec};
+use crate::tsm::reader::TSMReader;
```
Comment (Subsegment) [inline tskv/src/reader/column_group/mod.rs:23]: **TSM**Reader **Tsm**Writer
emmmmmm

Comment (Subsegment) [reply tskv/src/reader/column_group/mod.rs:23]: TSMReader --> TsmReader

Comment (liikeuforever) [reply tskv/src/reader/column_group/mod.rs:23]: In order to perform harmonization, it is a good solution to modify all Tsm to TSM

Comment (liikeuforever) [reply tskv/src/reader/column_group/mod.rs:23]: In order to perform harmonization, it is a good solution to modify all TSM to Tsm

**Pull request closed without merging**